main/test/ca/uwaterloo/flix/library/TestIterator.flix

```diff
@@ -1,5 +1,57 @@
 namespace TestIterator {
 
+    /////////////////////////////////////////////////////////////////////////////
+    // head                                                                    //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def testHead01(): Bool & Impure =
+        (Nil: List[Int32] |> Iterator.toIter |> Iterator.head) == None
+
+    @test
+    def testHead02(): Bool & Impure =
+        ((1 :: 2 :: Nil): List[Int32] |> Iterator.toIter |> Iterator.head) == Some(1)
+
+
+    /////////////////////////////////////////////////////////////////////////////
+    // range                                                                   //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def range01(): Bool & Impure =
+        Iterator.range(0, 0) |> Iterator.isEmpty
+
+    @test
+    def range02(): Bool & Impure =
+        Iterator.range(10, 0) |> Iterator.isEmpty
+
+    @test
+    def range03(): Bool & Impure =
+        let iter = Iterator.range(1, 4);
+        let one = Iterator.head(iter);
+        let two = Iterator.head(iter);
+        let three = Iterator.head(iter);
+        Some(1) == one and
+        Some(2) == two and
+        Some(3) == three and
+        None == Iterator.head(iter)
+
+    @test
+    def range04(): Bool & Impure =
+        let iter = Iterator.range(-1, 2);
+        let minusOne = Iterator.head(iter);
+        let zero = Iterator.head(iter);
+        let one = Iterator.head(iter);
+        Some(-1) == minusOne and
+        Some(0) == zero and
+        Some(1) == one and
+        None == Iterator.head(iter)
```

Comment:
I think these tests are too complicated.

It is okay to just to `Iter.range(1, 1) |> Iter.toList == theListYouExpect`

Reply:
Fixed it! :)